### ISSUE TITLE
Error casting numpy.core.Scalar to java.lang.Number in gradient boosting regressor learning rate 

### ISSUE THREAD

**Issue by moteleolu** [jpmml/jpmml-sklearn#96]
@vruusmann Related to what you said [here](https://groups.google.com/d/msg/jpmml/WZIpILIyaao/AxHHCabmAQAJ), and fixed [here](https://github.com/jpmml/jpmml-sklearn/commit/9192e0ca4e083442502ab103ea6a41c11a7de86f) for nearest neighbors 
when I'm working with `GradientBoostingTrees` and `BayesSearchCV`, the learning rate gets wrapped as a numpy scalar and encoding the model fails. 

```
SEVERE: Failed to convert
java.lang.ClassCastException: numpy.core.Scalar cannot be cast to java.lang.Number
	at sklearn.ensemble.gradient_boosting.GradientBoostingRegressor.getLearningRate(GradientBoostingRegressor.java:67)
	at sklearn.ensemble.gradient_boosting.GradientBoostingRegressor.encodeModel(GradientBoostingRegressor.java:59)
	at sklearn.ensemble.gradient_boosting.GradientBoostingRegressor.encodeModel(GradientBoostingRegressor.java:32)
	at sklearn2pmml.pipeline.PMMLPipeline.encodePMML(PMMLPipeline.java:215)
	at org.jpmml.sklearn.Main.run(Main.java:145)
	at org.jpmml.sklearn.Main.main(Main.java:94)
```

**Comment by vruusmann** [comment]
My integration testing code uses `GradientBoostingRegressor` directly, and the value of `learning_rate` attribute shows up on the Java side as `java.lang.Number`.

The transformation from `Number` to Numpy scalar must be happening because of `BayesSearchCV`.

Needs more systematic investigation and fixing.
